feat(curriculum): Add fCC Authors Page Project to JS Cert (#49655)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-fetch-and-promises-by-building-an-fcc-authors-page/63ff5850cd098f1796b5e9fa.md

---
id: 63ff5850cd098f1796b5e9fa
title: Step 3
challengeType: 0
dashedName: step-3
---

# --description--

When you make a `GET` request to an endpoint, the data you get is not always usable at first. To make it usable, you can use the `json()` method on the response with the `.then()` method.

> The `.then()` method can take a callback function for the fulfilled and rejected cases of the `Promise`.

Reviewer: I think we should add another example here for campers

```js
fetch("sample-endpoint-goes-here")
  .then((res) => res.json())
```

Author: Thanks. I think that would make it clearer. I've done it.

> Add a `.then()` method right under your `fetch()`. Inside the `.then()` method, name the response from the endpoint `res`, add a fat arrow for the callback, then use the `json()` method on that `res`. You don't need to use curly braces because the arrow function will have one expression. 

Reviewer: I think we should the work chain, to help reinforce that concept to campers. Maybe something like this. 

```suggestion
Chain the `.then()` method to your fetch call. Inside the `.then()` method, add a callback function with a `res` parameter that returns `res.json()`. You don't need to use curly braces because the arrow function will have one expression. 
```

---
id: 63ff5850cd098f1796b5e9fa
title: Step 3
challengeType: 0
dashedName: step-3
---

# --description--

> The `fetch()` method returns a `Promise`, which is  placeholder object that either resolves to a `Response` object, or is rejected. You can use the `.then()` method to access the `Promise` if it fulfils and the `.catch()` method to handle errors when it rejects.
> 
> You can attach a `.then()` to the `fetch()` method this way:
> 
> ```js
> fetch("sample-url-goes-here")
>   .then((res) => console.log(res))
> ```
> 
> Chain the `.then()` method to your fetch call. Inside the `.then()` method, add a callback function with a `res` parameter, then log the `res` to the console so you can see the `Response` object. Open your console and expand the `Response` object to see what it contains.
> 
> Again, don't terminate the code with a semi-colon yet.
> 
> **Note**: You can open up the built-in freeCodeCamp console or your browser's console to see the data you're getting. Again, don't terminate your code with a semi-colon yet.

Reviewer 2: I tried to shorten this a bit, and remove concepts that will be taught later:

``````suggestion
The `fetch()` method returns a `Promise`, which is a placeholder object that will either be fulfilled if your request is successful, or rejected if your request is unsuccessful.

If the `Promise` is fulfilled, it resolves to a `Response` object, and you can use the `.then()` method to access the `Response`.

Here's how you can chain `.then()` to the `fetch()` method:

```js
fetch("sample-url-goes-here")
  .then((res) => console.log(res))
```

Chain the `.then()` method to your `fetch` call. Inside the `.then()` method, add a callback function with a `res` parameter, then log the `res` to the console so you can see the `Response` object. Open your console and expand the `Response` object to see what it contains.

Again, don't terminate the code with a semi-colon yet.

**Note**: You can open up the built-in freeCodeCamp console or your browser's console to see the data you're getting.
``````
